### ISSUE TITLE
[Question] 标签一定要str类型吗，可以拿来做情感分析吗

### ISSUE THREAD

**Issue by Owenscu** [BrikerMan/Kashgari#27]
比如AI Challenger2018情感分析的多标签怎么处理喂进去

**Comment by BrikerMan** [comment]
多标签分类的模型还没有增加，可以提供详细的数据集链接，我会看情况增加相关任务。

**Comment by Owenscu** [comment]
@BrikerMan 这是数据集链接 链接：https://pan.baidu.com/s/1spbM8QuyjRmgCSa9i-P7Mw 
提取码：679y 
只有7天有效哦

**Comment by BrikerMan** [comment]
这个问题并不是单个维度的多标签，感觉需要专门建模处理才可以，我这里能支持的是 `[(1, 2), (3,4),(5,)]` 这种简单的多标签多分类。

**Comment by BrikerMan** [comment]
这样的多分类能满足需求嘛，y 是由多个标签组成的~

**Comment by alexwwang** [comment]
A multi label classification problem could be solved by training a model with `sigmoid` as activation function and `binary-crossentropy` as loss function.  The output is an n-dim one-hot vector to predict n possible labels.
That's it.

**Comment by BrikerMan** [comment]
@alexwwang check this #29 out, is this looks good to you? I am planning to add multi_label classification, but it seems we had to change all the classification models to support this feature.

**Comment by alexwwang** [comment]
Is it possible to wrap up a nn models in a multi-label task model? I mean just replace the last output layer to fit with different kind of classification tasks and allow users to determine which to choose.

Meanwhile I am considering allowing users setting hyper_parameters while initializing a model class in the model zoo.

I think these two aspects could be put together, so the flexibility and concision could both be kept.

**Comment by BrikerMan** [comment]
I think your solution is better than mine, please make the changes and submit a pull request~ @alexwwang

**Comment by alexwwang** [comment]
Ok but I'm afraid it would take some time. Fighting the shape bug yet.

**Comment by BrikerMan** [comment]
Take your time, no need to rush.

**Comment by alexwwang** [comment]
#34 This commit fulfilled this need. By passing `sigmoid` and `binary_crossentropy` hyper_parameters to init function of a classification model, you could get one model support multi-label vector output.

**Comment by BrikerMan** [comment]
@alexwwang I think multi-label is not finished yet, still need to change here https://github.com/BrikerMan/Kashgari/blob/692e2c23f8400a73bb851ac888606942bc94b070/kashgari/tasks/classification/base_model.py#L110 to process multi_label.

**Comment by alexwwang** [comment]
@BrikerMan Yeah, the data padding work. The nearest approach, I think, maybe with the help of [sklearn.preprocessing.MultiLabelBinarizer](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html). And this tool could also deal with multi-class classification as a specific type of multi-label classification. Or just add up another switch to approach multi-label y-vector, leaving current part unchanged.
Whatever, seems there's no shortcut here in data preprocessing and prediction processing, if we want to keep the predict confidence/probability of each label/class.
How do you think?

**Comment by BrikerMan** [comment]
Please check out the #29, I have used `sklearn.preprocessing.MultiLabelBinarizer` and rewrite the data-process and predict function.

**Comment by BrikerMan** [comment]
Kashgari supported multi-label classification now, change y from string to a string list, then set `multi_label=True` While init model class.
